src/walletSdk/Anchor/Sep6.ts

```diff
@@ -53,4 +60,79 @@ export class Sep6 {
       throw new ServerRequestFailedError(e);
     }
   }
+
+  /**
+   * Deposits funds using the SEP-6 protocol. Next steps by
+   * the anchor are given in the response.
+   *
+   * @param {object} options - The options for the deposit.
+   * @param {string} options.authToken - The authentication token.
+   * @param {Sep6DepositParams} options.params - The parameters for the deposit request.
+   *
+   * @returns {Promise<Sep6DepositResponse>} Sep6 deposit response, containing next steps if needed
+   * to complete the deposit.
+   *
+   * @throws {Sep12MissingInfoError} If Sep-12 KYC info is missing for the user.
+   * @throws {Sep6DepositDeniedError} If the deposit is still being processed or not accepted.
+   * @throws {Error} If an unexpected error occurs during the deposit operation.
+   */
+  async deposit({
+    authToken,
+    params,
+  }: {
+    authToken: string;
+    params: Sep6DepositParams;
+  }): Promise<Sep6DepositResponse> {
+    const { transferServer } = await this.anchor.sep1();
+
+    try {
+      const resp = await this.httpClient.get(
+        `${transferServer}/deposit?${queryString.stringify(params)}`,
+        {
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${authToken}`,
+          },
+        },
+      );
+      return resp.data;
+    } catch (e) {
+      if (e.response?.data?.type === "non_interactive_customer_info_needed") {
+        throw new Sep12MissingInfoError(e.response?.data?.fields);
```

Comment:
I don't think it's an error. It's expected behavior in the flow. I think we should have different types of the Sep6Response which wallets must handle accordingly

Reply:
BTW you can find flow diagrams for SEP-6 here
https://github.com/stellar/java-stellar-anchor-sdk/tree/develop/docs/diagrams/sep6

Reply:
good point, how about this? https://github.com/stellar/typescript-wallet-sdk/pull/77/commits/fb25d34c7e844d11cb12ac976eedb77ec9dd441c

also I realized those resposnes can be for deposit or withdrawal so i added a helper flow function similar to sep24